src/uu/tr/src/tr.rs

```diff
@@ -139,17 +133,10 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             let op = SqueezeOperation::new(set1, complement_flag);
             translate_input(&mut locked_stdin, &mut buffered_stdout, op);
         } else {
-            let mut translate_buffer = vec![];
-            {
-                let mut writer = BufWriter::new(&mut translate_buffer);
-                let op = TranslateOperation::new(set1, set2.clone(), complement_flag)?;
-                translate_input(&mut locked_stdin, &mut writer, op);
-            }
-            {
-                let mut reader = BufReader::new(translate_buffer.as_bytes());
-                let squeeze_op = SqueezeOperation::new(set2, false);
-                translate_input(&mut reader, &mut buffered_stdout, squeeze_op);
-            }
+            let translate_op = TranslateOperation::new(set1, set2.clone(), complement_flag)?;
```

Comment:
As you are working on it and understand the code, it would be nice to add some comments explaining what is doing because it is a bit cryptic if you aren't familiar with this :)

Reply:
Apparently you're right in that I understand the code, because I don't see the crypticness :D

Can you clarify which aspect is cryptic?

- `translate_op` and `squeeze_op` are the two transformations that are selected, but the names already say that.
- They must be applied in this order, but `chain` already says that.
- `SymbolTranslator::chain` does pretty much the same thing as [`Iterator::chain`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.chain), would a link to it help?
- And finally we transform the entire input with the combined operation, but the code already says that.

Reply:
Okay! I'll mostly copy-paste the above description.